server/service/src/vaccination/update/generate.rs

```diff
@@ -73,8 +79,11 @@ fn generate_given(
 
     // apply given status, stock and invoice ids
     let vaccination = VaccinationRow {
+        not_given_reason: None,
         given: true,
+        given_store_id: Some(store_id.to_string()),
```

Comment:
The bit that actually solves the PR, assigning given store ID where we mark as given - and in validate method, checking this matches the calling store id